### PR TITLE
Migrate from failure to anyhow and thiserror

### PR DESCRIPTION
**Pull request by andor0**
Replace rust failure library with anyhow and thiserror. (Issue #689)

### CODE REVIEW

File: server/src/apub/extensions/signatures.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@ use crate::{apub::ActorType, LemmyError};
 use activitystreams_ext::UnparsedExtension;
 use activitystreams_new::unparsed::UnparsedMutExt;
 use actix_web::{client::ClientRequest, HttpRequest};
+use anyhow::{anyhow as format_err};
```
Comment (dessalines) [inline server/src/apub/extensions/signatures.rs:5]: Could you change these to just `anyhow::anyhow`, and change all the `format_err!` to `anyhow!` below? thanks.

**Comment by dessalines** [comment]
Looks good! I'll test now.

**Comment by dessalines** [comment]
Just tested, looks good to me. @Nutomic if you'd take a quick look too before merge.

**Comment by Nutomic** [comment]
Looks good, thank you @andor0 :)

Btw @dessalines I noticed that test_espeak is failing on main.

**Comment by dessalines** [comment]
I tested on mine, it passed. Its probably because you don't have `espeak` installed on your system.

**Comment by Nutomic** [comment]
Ah makes sense. But in that case, you should add it in the docs ;)

https://github.com/LemmyNet/lemmy/blob/main/docs/src/contributing_local_development.md

**Comment by dessalines** [comment]
K, I added this to the ubuntu install.

**Pull request merged**